src/wasm_offset_converter.js

```diff
@@ -0,0 +1,82 @@
+function WasmOffsetConverter(wasmBytes) {
+  // This class parses a WASM binary file, and constructs a mapping from
+  // function indices to the start of their code in the binary file.
+  //
+  // The main purpose of this module is to enable the conversion of function
+  // index and offset from start of function to an offset into the WASM binary.
+  // This is needed to look up the WASM source map as well as generate
+  // consistent program counter representations given v8's non-standard
+  // WASM stack trace format.
```

Comment:
please add a link to the v8 bug, or something else explaining that. that way we can remove this when v8 fixes it. (or can we remove it? sounds like we still need it anyhow for source maps?)

Reply:
The #8711 (LSan) ended up using this metadata for source maps (for now, at least) to make sure that the binary search doesn't end up with an offset outside of the function. But the current incoming branch does not need this on fixed v8 or spidermonkey. We'll still need this for a while after v8 fixes it, as the ancient versions of node become abandoned.